i3bar/src/outputs.c

```diff
@@ -195,8 +195,8 @@ static int outputs_end_map_cb(void *params_) {
         bool handle_output = false;
         for (int c = 0; c < config.num_outputs; c++) {
             if (strcasecmp(params->outputs_walk->name, config.outputs[c]) == 0 ||
-                (strcasecmp(config.outputs[c], "primary") == 0 &&
-                 params->outputs_walk->primary)) {
+                (strcasecmp(config.outputs[c], "primary") == 0 && params->outputs_walk->primary) ||
+                (strcasecmp(config.outputs[c], "nonprimary") == 0 && !params->outputs_walk->primary)) {
```

Comment:
I think this will be  a bit more readable with a const:
```c
const bool is_primary = params->outputs_walk->primary
```

```suggestion
                (strcasecmp(config.outputs[c], "primary") == 0 && is_primary ||
                (strcasecmp(config.outputs[c], "nonprimary") == 0 && !is_primary)) {
```